### ISSUE TITLE
bug: Pasting with vim-pasta results in corrupted undo

### ISSUE THREAD

**Issue by trkoch** [vscode-neovim/vscode-neovim#2156]
### Check the following:

- [X] I have checked the behavior after setting "vscode-neovim.neovimClean" in VSCode settings.
- [X] I have read the vscode-neovim docs.
- [X] I have searched existing issues.

### Neovim version (nvim -v)

NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1716656478 

### Operating system/version

macOS 14.5

### Describe the bug

Pasting with [vim-pasta](https://github.com/ku1ik/vim-pasta/tree/master) cannot be completely undone with `u`. It used to work up until `v1.17.2`. I tracked down the issue (using `git bisect`) to 61f4e4f842d7a7a559b60e7e82f3acb26c908abb introduced with https://github.com/vscode-neovim/vscode-neovim/pull/2127.

In other words, `vim-pasta` appears to [modify](https://github.com/ku1ik/vim-pasta/blob/2b786703eef9f82ae7a56f3de4ee43e1e5efaaa5/plugin/pasta.vim#L11-L21) the buffer such that the changes cannot be undone cleanly by pressing `u` until no more recorded changes exist. I suspect this bug is not strictly related to the plugin, but so far I have not been able to reproduce it without `vim-pasta`.

### Steps To Reproduce

1. Install vim-pasta
2. Yank a couple of lines
3. Paste x times
4. Undo x times

### Expected Behavior

All changes are reverted. The statusline reads "Already at oldest change". The dirty indicator is not visible.

**Comment by trkoch** [comment]
`"editor.formatOnSave": true,` seems to be another prerequisite.

**Comment by xiyaowong** [comment]
It's unlikely that this will be fixed (you can only modify "editor.formatOnSave"), but you can try a more modern plugin, `gbprod/yanky.nvim` (Implemented using the nvim API, it results in fewer changes.)

You can subscribe to: https://github.com/microsoft/vscode/issues/221614
